parser/formattext.go

```diff
@@ -98,17 +97,21 @@ func (fc *FontConfig) FormatText(text string, maxWidth int, cursorOverlapWidth i
 			// it could span multiple words. The true solution would require optimistically trying to fit all
 			// remaining words onto the same line, rather than only looking at the current word + cursor. However,
 			// this is "good enough" and likely works for almost all actual use cases in practice.
-			if len(nextWord) > 0 && (!isFirstLine || fc.isParagraphBreak(nextWord)) {
+			if len(nextWord) > 0 && (fc.isFirstLine(lineNumber) == false || fc.isParagraphBreak(nextWord)) {
 				nextWidth += cursorOverlapWidth
 			}
 			if nextWidth > maxWidth && curLineSb.Len() > 0 {
 				formattedSb.WriteString(curLineSb.String())
-				if isFirstLine {
-					formattedSb.WriteString(`\n`)
-					isFirstLine = false
-				} else {
+
+                if fc.isFirstLine(lineNumber) {
+                    formattedSb.WriteString(`\n`)
+				} else if fc.isMaxLineOrGreater(lineNumber, NumLines) {
 					formattedSb.WriteString(`\l`)
-				}
+				} else {
+                    formattedSb.WriteString(`\n`)
+                }
+
+                lineNumber++
```

Comment:
We can simplify this block to something like:
```go
if fc.isMaxLineOrGreater(lineNumber, NumLines) {
    formattedSb.WriteString(`\l`)
} else {
    formattedSb.WriteString(`\n`)
}
```

Reply:
https://github.com/huderlem/poryscript/pull/59/commits/9296e060f71101062e2515c6711497458c60ac12